packages/utilities/src/selection/Selection.ts

```diff
@@ -191,6 +207,14 @@ export class Selection<TItem = IObjectWithKey> implements ISelection<TItem> {
       hasSelectionChanged = true;
     }
 
+    if (!haveItemsChanged) {
+      for (const key of Object.keys(this._keyToIndexMap)) {
+        if (!(key in newKeyToIndexMap)) {
+          haveItemsChanged = true;
```

Comment:
Once you've found that 1 item has changed, you should be able to break the loop right? No need to continue looping through all 10000 keys in the map.

Reply:
Ah, yes, good point.